### PR TITLE
Add --grep, --invert, --bail only <filenames>+ to traceurTests.js

### PR DESCRIPTION
**Pull request by johnjbarton**
The -- options call mocha equivalents.
'only' supports focused testing.
Rename test/modular/tests.js to test/traceurTests.js

### CODE REVIEW

**Comment by johnjbarton** [comment]
depends on #1726

**Comment by johnjbarton** [comment]
PTAL

File: test/traceurTests.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    'test/unit/runtime/*.js',
+    'test/unit/system/*.js',
+    'test/unit/node/*.js',
+    'test/unit/*.js'
+  ];
+  return {patterns: patterns};
+}
+
+function mergeWithOptions(testOptions) {
+
```
Comment (arv) [inline test/traceurTests.js]: remove empty line

File: test/traceurTests.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  let commandLine =
+      new commander.Command(process.argv[0] + ' ' + process.argv[1]);
+
+  for(let prop in testOptions)
```
Comment (arv) [inline test/traceurTests.js]: ws after for

Comment (arv) [inline test/traceurTests.js]: Yuk. for-in loop.

Comment (arv) [inline test/traceurTests.js]: Missing {}

File: test/traceurTests.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    commandLine[prop] = testOptions[prop];
+
+  commandLine.option('-?, --help', 'Show this help text', () =>{
+    commandLine.help();
+  }).usage('[options] [files]')
+
+  // Selected mocha options supported.
+  commandLine.option('-g, --grep <pattern>', 'only run tests matching <pattern>').
+      option('-i, --invert', 'inverts --grep matches').
+      option('-b, --bail', "bail after first test failure");
```
Comment (arv) [inline test/traceurTests.js]: why "?

File: test/traceurTests.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  
+  commandLine.command('only <file> [files...]').
```
Comment (arv) [inline test/traceurTests.js]: Is it `only` or `--only`?

File: test/traceurTests.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      description('only test these [files] ').action(
+      (file, files) => {
+        commandLine.patterns = [file];
+        if (files)
```
Comment (arv) [inline test/traceurTests.js]: {}

File: test/traceurTests.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    runner.on('end', () => {
+      process.exit(failed);
+    });
+  });
+}
+
+runTests(
+    mergeWithOptions(
+        defaultTestOptions()
+    )
```
Comment (arv) [inline test/traceurTests.js]: ) on previous line

**Comment by arv** [comment]
LGTM

File: test/traceurTests.js (hunk 1 above)
Comment (johnjbarton) [inline test/traceurTests.js]: done

File: test/traceurTests.js (hunk 2 above)
Comment (johnjbarton) [inline test/traceurTests.js]: Object.getOwnPropertyNames used. I was wondering if I've ever used for-in.

File: test/traceurTests.js (hunk 3 above)
Comment (johnjbarton) [inline test/traceurTests.js]: done. copypasta.

File: test/traceurTests.js (hunk 4 above)
Comment (johnjbarton) [inline test/traceurTests.js]: `only`, a 'sub-command', eg `./tval only test/util/url.js`

File: test/traceurTests.js (hunk 5 above)
Comment (johnjbarton) [inline test/traceurTests.js]: ? 

We have quite a lot of 
```
  if (cond) 
    statement;
```
I removed a blank line.

File: test/traceurTests.js (hunk 6 above)
Comment (johnjbarton) [inline test/traceurTests.js]: but not the `);`?

File: test/traceurTests.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    runner.on('fail', (err) => {
+      failed++;
+    });
+    runner.on('end', () => {
+      process.exit(failed);
+    });
+  });
+}
+
+runTests(
```
Comment (arv) [inline test/traceurTests.js]: I would just write this as:

```js
runTests(mergeWithOptions(defaultTestOptions()));
```

File: test/traceurTests.js (hunk 5 above)
Comment (arv) [inline test/traceurTests.js]: Yeah, I realized that  after commenting... I think we should go back to always requiring {} but I'll try to not comment on these.

**Pull request closed without merging**